lib/internal/modules/esm/translators.js

```diff
@@ -156,14 +179,13 @@ translators.set('json', async function jsonStrategy(url) {
 
 // Strategy for loading a wasm module
 translators.set('wasm', async function(url) {
-  const pathname = fileURLToPath(url);
-  const buffer = await readFileAsync(pathname);
+  const buffer = await getSource(url);
   debug(`Translating WASMModule ${url}`);
   let compiled;
   try {
     compiled = await WebAssembly.compile(buffer);
   } catch (err) {
-    err.message = pathname + ': ' + err.message;
+    err.message = url + ': ' + err.message;
```

Comment:
Can we keep this to `pathname`? I think it makes error messages shorter and also simpler for users.

Reply:
IDK, What should it be when it isn't a `file:` URL?

Reply:
Ah right. Perhaps we can have a convention that if it is a `file:` URL we display it as a path, but if it is any other scheme we display the full URL? I'd just like us to be consistent and I think removing the leading `file:` provides a better user experience in places where it doesn't add any new information.

Reply:
We can remove the extra bits for files but we do lose some identity info from fragment/query

Reply:
Thanks, I was just going through the error messages recently and was finding it a little inconsistent that we sometimes give the path and sometimes the URL. The main thing I think is that we have a convention and stick to it. This one seems the most usable to me, but if there are concerns in future we could always switch to full URLs everywhere.